src/test/java/org/folio/rest/api/HridSettingsIncreaseMaxValueMigrationTest.java

```diff
@@ -0,0 +1,139 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.api.StorageTestSuite.getVertx;
+
+import java.util.List;
+
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.ResourceUtil;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class HridSettingsIncreaseMaxValueMigrationTest extends TestBase {
```

Comment:
These tests are a very different style to the other tests. Please can you help me understand why this approach was chosen?

In particular it appears to be binding closer to vert.x futures, which are getting a significant change in vert.x 4.0

Reply:
I chose the async style because there are a lot of DB operations that have to be executed to create/destroy/get next value/set value, which are async.

If you think that this approach may cause compatibility issues, then I think I can convert it to the blocking style.

Reply:
Ok, I personally find these kinds of tests, with the vert.x context and chaining to not be very expressive. And I reckon they will need changing soon.

I suggest we leave this one as is, as an isolated case. Maybe we could add a comment to describe the reasoning for this style.

To hide the asynchronous pain, I've typically extracted methods that block.

Reply:
I've refactored it to be blocking.